src/robot/output/console/verbose.py

```diff
@@ -119,40 +155,77 @@ def status(self, status, clear=False):
-    def _clear_status(self):
-        self._clear_info()
-        self._write_info()
+    def _clear_status(self, erase_last_message=True):
+        self._clear_debug()
+        self._write_debug(erase_last_message)
 
-    def _clear_info(self):
+    def _clear_debug(self):
         self._stdout.write('\r%s\r' % (' ' * self._width))
         self._keyword_marker.reset_count()
 
     def message(self, message):
         if message:
             self._stdout.write(message.strip() + '\n')
 
-    def keyword_marker(self, status):
-        if self._keyword_marker.marker_count == self._status_length:
-            self._clear_status()
-            self._keyword_marker.reset_count()
-        self._keyword_marker.mark(status)
+    def start_keyword_marker(self, kw, depth):
+        if not self._keyword_marker.marking_enabled:
+            return
+        if depth <= self._verbose_level:
+            name = self._keyword_marker.label_writer(kw)
+            label = self._keyword_label(name, depth)
+            self._start_keyword_marker(label)
+
+    def end_keyword_marker(self, kw, depth):
+        if not self._keyword_marker.marking_enabled:
+            return
+        status = 'PASS' if kw.status != 'FAIL' else 'FAIL'
+        if depth <= self._verbose_level:
+            self._end_keyword_marker(status)
+        else:
+            if self._keyword_marker.marker_count == self._status_length:
+                self._clear_status(False)
+                self._keyword_marker.reset_count()
+            self._keyword_marker.mark(status)
+
+    def _start_keyword_marker(self, label):
+        if self._break_line_needed:
+            self._stdout.write('\n')
+        self.debug(label, None)
+        self._break_line_needed = True
+
+    def _end_keyword_marker(self, status):
+        self.status(status, clear=True)
+        self._break_line_needed = False
+
+    def _keyword_label(self, name, depth):
+        width, separator = self._get_width_and_separator(False)
+        prefix_str = separator * 4 * depth
+        return '%s%s' % (prefix_str, name)
 
     def error(self, message, level, clear=False):
         if self._should_clear_markers(clear):
-            self._clear_info()
+            self._clear_debug()
         self._stderr.error(message, level)
         if self._should_clear_markers(clear):
-            self._write_info()
+            self._write_debug()
 
     def output(self, name, path):
         self._stdout.write('%-8s %s\n' % (name+':', path))
 
+    def increase_verbose_level(self, level=1):
+        self._verbose_level += level
+
+    def decrease_verbose_level(self, level=1):
+        self.increase_verbose_level(-level)
+        assert( self._verbose_level >= 0 )
+
 
 class KeywordMarker(object):
 
     def __init__(self, highlighter, markers):
         self._highlighter = highlighter
         self.marking_enabled = self._marking_enabled(markers, highlighter)
+        self.label_writer = lambda kw: "%s  %s" % (kw.name, "  ".join(kw.args))
```

Comment:
args can be other than string, when called via BuiltIn().run_keyword(). 
This will fail in that case.